swap/src/bob/swap.rs

```diff
@@ -390,15 +390,28 @@ where
                 .await
             }
             BobState::Cancelled(state) => {
-                // TODO
                 // Bob has cancelled the swap
                 let state = match state.current_epoch(bitcoin_wallet.as_ref()).await? {
                     Epoch::T0 => panic!("Cancelled before t1??? Something is really wrong"),
                     Epoch::T1 => {
-                        state.refund_btc(bitcoin_wallet.as_ref()).await?;
+                        // If T1 has expired but not T2 we must be able to refund
+                        state.submit_tx_refund(bitcoin_wallet.as_ref()).await?;
                         BobState::BtcRefunded(state)
                     }
-                    Epoch::T2 => BobState::Punished,
+                    Epoch::T2 => {
+                        // If T2 expired we still try to refund in case Alice has not acted.
+                        // This is especially important for scenarios where Alice is offline
+                        // indefinitely after starting the swap.
+                        info!("Trying to refund even though T2 has passed...");
+                        let refund_result = state.submit_tx_refund(bitcoin_wallet.as_ref()).await;
+                        match refund_result {
+                            Ok(()) => BobState::BtcRefunded(state),
+                            Err(err) => {
+                                warn!("Bob tried to refund after T2 has passed, but sending the refund tx errored with {}", err);
```

Comment:
```suggestion
                                error!("Bob tried to refund after T2 has passed, but sending the refund tx errored with {}", err);
```
This is a pretty bad scenario.